### ISSUE TITLE
firmware for arduino mega 2560 does not compile ( v 2.0)

### ISSUE THREAD

**Issue by Goodfeat** [repetier/Repetier-Firmware#905]
Hello, I decided to try firmware 2.0 again. but I still can't even compile it.

I did according to the instructions

![image](https://user-images.githubusercontent.com/44455128/74924517-5fba0500-541e-11ea-8ed9-b53cb5d80d5d.png)

changed the motherboard to ramps
![image](https://user-images.githubusercontent.com/44455128/74924654-8bd58600-541e-11ea-96c9-d7669950015c.png)

added a line to the constant.x file
![image](https://user-images.githubusercontent.com/44455128/74924782-c2ab9c00-541e-11ea-95e0-48eaaea189de.png)

It seems that this should be enough for the firmware to compile so that I test it.
but errors are streaming in the terminal and the firmware does not compile for the desired dress.
![image](https://user-images.githubusercontent.com/44455128/74924940-174f1700-541f-11ea-8665-29405c49bb78.png)

![image](https://user-images.githubusercontent.com/44455128/74924966-22a24280-541f-11ea-8c7e-8420b0308547.png)

please tell me why the firmware does not compile and how can I fix it?

**Comment by repetier** [comment]
The 8 bit HAL required for HAL is not finished. It is already there from old V1 but not adopted to new requirements. Currently only due based boards, RUMBA32 and Adafruid Grand Central M4 +Board are supported and will compile. AVR will take a bit more time.

**Comment by Goodfeat** [comment]
Does this mean it's useless to install firmware on arduino mega now?
I'm sorry, I was set up to test the new firmware.

**Comment by repetier** [comment]
Yes, it is not compilable for AVR at the moment. We just have started adding the files but not finished developing them. That is why you see them already but can not compile.

**Comment by Goodfeat** [comment]
OK

I hope it will work soon
I will wait for the firmware update
